catalyst_voices/lib/pages/registration/create_keychain/stage/seed_phrase_check_panel.dart

```diff
@@ -85,40 +41,72 @@ class _SeedPhraseCheckPanelState extends State<SeedPhraseCheckPanel> {
   }
 
   void _clearUserWords() {
-    setState(_updateUserWords);
+    RegistrationCubit.of(context).keychainCreation.setUserSeedPhraseWords([]);
   }
 
   void _onWordsSequenceChanged(List<String> words) {
-    setState(() {
-      _updateUserWords(words);
-    });
+    RegistrationCubit.of(context)
+        .keychainCreation
+        .setUserSeedPhraseWords(words);
   }
+}
 
-  void _updateSeedPhraseWords() {
-    final seedPhrase = widget.seedPhrase;
-    final words = seedPhrase?.mnemonicWords ?? <String>[];
-    final shuffledWords = seedPhrase?.shuffledMnemonicWords ?? <String>[];
+class _BlocLoadable extends StatelessWidget {
+  final WidgetBuilder builder;
 
-    _seedPhraseWords
-      ..clear()
-      ..addAll(words);
+  const _BlocLoadable({
+    required this.builder,
+  });
 
-    _shuffledSeedPhraseWords
-      ..clear()
-      ..addAll(shuffledWords);
+  @override
+  Widget build(BuildContext context) {
+    return BlocSeedPhraseBuilder<bool>(
+      selector: (state) => state.isLoading,
+      builder: (context, state) {
+        return VoicesLoadable(
+          isLoading: state,
+          builder: builder,
+        );
+      },
+    );
   }
+}
 
-  void _updateUserWords([
-    List<String> words = const [],
-  ]) {
-    _userWords
-      ..clear()
-      ..addAll(words);
+class _BlocSeedPhraseWords extends StatelessWidget {
+  const _BlocSeedPhraseWords({
+    required this.onUserWordsChanged,
+    this.onUploadTap,
+    this.onResetTap,
+  });
 
-    final isConfirmed = _completedCorrectlyWordsSequence;
+  final ValueChanged<List<String>> onUserWordsChanged;
+  final VoidCallback? onUploadTap;
+  final VoidCallback? onResetTap;
 
-    RegistrationCubit.of(context)
-        .setSeedPhraseCheckConfirmed(isConfirmed: isConfirmed);
+  @override
+  Widget build(BuildContext context) {
+    return BlocSeedPhraseBuilder<
+        (
+          List<String> shuffledWords,
+          List<String> words,
+          bool isResetWordsEnabled,
+        )>(
+      selector: (state) => (
+        state.shuffledWords,
+        state.userWords,
+        state.isResetWordsEnabled,
+      ),
+      builder: (context, state) {
+        return _SeedPhraseWords(
+          words: state.$1,
```

Comment:
Is it possible to use named params for the record? There are two params that have the same type (List<String>), quite easy to do a mistake